src/Units/ErrorMarker.vala

```diff
@@ -39,6 +39,8 @@ namespace Units {
         if (error.source == null)
           return;
         string myfilename = error.source.file.get_relative_filename();
+        if (myfilename == null)
```

Comment:
Does this case actually occur? I thought that once the error.source exists, it should also have a corresponding file?